pkgs/applications/kde/kdenlive/default.nix

```diff
@@ -80,13 +82,14 @@ mkDerivation {
     kpurpose
     kdeclarative
     wrapGAppsHook
-  ];
+  ] ++ lib.optional enableGlaxnimate glaxnimate;
   # Both MLT and FFMpeg paths must be set or Kdenlive will complain that it
   # doesn't find them. See:
   # https://github.com/NixOS/nixpkgs/issues/83885
-  patches = [ ./dependency-paths.patch ];
+  patches = [ ./dependency-paths.patch ] ++ lib.optional enableGlaxnimate ./dependency-paths-glaxnimate.patch;
 
-  inherit mediainfo;
+  # Needed to replace @glaxnimate@ by its path
+  inherit mediainfo glaxnimate;
```

Comment:
Doesn't this need to be done with substituteAll on the patch file?

Reply:
Sure, but `substituteAllInPlace` is already part of Kdenlive's derivation as this file is already patched similarly for other binaries.

Reply:
Can we make this explicit? Using substituteAllInPlace with ambient variables is kinda surpising.

Reply:
What do you mean? substituteAllInPlace just expects these variables to be in the environment (whether this is good or not is discussed [here](https://github.com/NixOS/nixpkgs/issues/237216), but so far no agreement has been reached), and my goal was just to implement this fix following what kdenlive's package is already doing, as I don't want to risk adding bugs doing some sort of refactoring. Or do you mean to put them in `env = {}` to make it somewhat clearer?

Reply:
I agree, it would be cleaner to use the substituteAll builder on the patch itself. That said, I don't mind either way.

Reply:
I added an `env` in case you find it clearer, hopefully it will not introduce side effects (but after a quick test it seems to work). Any larger refactoring should be done in a different PR.